packages/bappo-components/src/primitives/ActivityIndicator/ActivityIndicator.web.tsx

```diff
@@ -0,0 +1,102 @@
+import { VisibilityProperty } from 'csstype';
+import * as React from 'react';
+import styled, { css, keyframes } from 'styled-components';
+
+import { DivViewBase } from '../../internals/web/ViewBase';
+import { ActivityIndicatorProps } from './types';
+
+const sizeMap = {
+  small: 20,
+  large: 36,
+};
+
+type Props = ActivityIndicatorProps & {
+  className?: string;
+};
+
+export default function ActivityIndicator({
+  animating = true,
+  color = '#999',
+  size = 'small',
+  style,
+  className,
+  testID,
+}: Props) {
+  const containerStyleProps = {
+    className,
+    style,
+    testID,
+  };
+  const spinnerContainerStyleProps = {
+    $size: sizeMap[size],
+  };
+  const spinnerStyleProps = {
+    $color: color,
+  };
+
+  const containerStyle: { visibility?: VisibilityProperty } = {};
+  if (!animating) containerStyle.visibility = 'hidden';
+
+  return (
+    <Container {...containerStyleProps} style={containerStyle}>
```

Comment:
I suggest just pass testID to container instead of put it in containerStyleProps - the naming causes confusion.

Reply:
Thanks for this, I've specified it as a separate prop instead of incorrectly using `containerStyleProps`